integration-tests/src/test/java/tachyon/master/JournalIntegrationTest.java

```diff
@@ -394,6 +397,42 @@ private void folderTestUtil(FileInfo fileInfo)
     fsMaster.stop();
   }
 
+  @Test
```

Comment:
This tests the scenario where we first create a directory, say "/a/b/", with NO_PERSIST, then create a file, say "/a/b/c" with SYNC_PERSIST, in that case, "/a/b" needs to be persisted too.